backend/lib/deal-observer.js

```diff
@@ -25,18 +26,26 @@ export async function observeBuiltinActorEvents (blockHeight, pgPool, makeRpcReq
  */
 export async function fetchDealWithHighestActivatedEpoch (pgPool) {
   const query = 'SELECT * FROM active_deals ORDER BY activated_at_epoch DESC LIMIT 1'
-  const { rows } = await pgPool.query(query)
-  const result = rows.map((row) => Value.Parse(ActiveDealDbEntry, row))
+  const result = (await pgPool.query(query)).rows.map((row) => {
+    const parsedRow = Value.Parse(ActiveDealDbEntry, row)
+    parsedRow.piece_cid = fromJSON(JSON.parse(parsedRow.piece_cid))
+    return parsedRow
+  })
   return result.length > 0 ? result[0] : null
 }
 
+/**
+ * @param {Array<BlockEvent>} activeDeals
+ * @param {Queryable} pgPool
+ * @returns {Promise<void>}
+ * */
 export async function storeActiveDeals (activeDeals, pgPool) {
   const transformedDeals = activeDeals.map((deal) => (
     {
       activated_at_epoch: deal.height,
       miner_id: deal.event.provider,
       client_id: deal.event.client,
-      piece_cid: deal.event.pieceCid,
+      piece_cid: toJSON(deal.event.pieceCid),
```

Comment:
Same comment here, why are we converting from `CID` instance to JSON?

If it's not practical to use the `CID` type (class), then let's use the string form (`baga...`) instead.